compiler/rustc_middle/src/thir/visit.rs

```diff
@@ -24,6 +25,8 @@ pub trait Visitor<'a, 'tcx: 'a>: Sized {
     fn visit_pat(&mut self, pat: &Pat<'tcx>) {
         walk_pat(self, pat);
     }
+
+    fn visit_constant(&mut self, _constant: ConstantKind<'tcx>) {}
```

Comment:
this looks like dead code to me

Reply:
We need this in the `is_poly` visitor in abstract-const creation.

Reply:
Yes, my idea is to directly check that in `visit_expr` or `visit_pat` instead of using `visit_constant`.

Considering that `visit_constant` doesn't actually visit most constants in the body, that method seems to me to pretty much only cause confusion